lib/mixlib/install/backend/bintray.rb

```diff
@@ -172,20 +191,71 @@ def parse_platform_info(artifact_map)
           }
         end
 
-        def bintray_get(resource)
-          uri = URI.parse(endpoint)
-          http = Net::HTTP.new(uri.host, uri.port)
-          http.use_ssl = (uri.scheme == "https")
-
-          path = "#{uri.path}/packages/chef/#{options.channel}/#{options.product_name}/#{resource}"
-          request = Net::HTTP::Get.new(path)
-          request.basic_auth(BINTRAY_USERNAME, BINTRAY_PASSWORD)
-
-          res = http.request(request)
-
-          # Raise if response is not 2XX
-          res.value
-          JSON.parse(res.body)
+        #
+        # Determines the architecture for which a file is published from from
+        # filename.
+        #
+        # We determine the architecture  based on the filename of the artifact
+        # since architecture the artifact is published for is not available
+        # in bintray.
+        #
+        # @param [String] filename
+        #
+        # @return [String]
+        #   one of the standardized architectures for Chef packages:
+        #   x86_64, i386, powerpc, sparc, ppc64, ppc64le
+        def parse_architecture_from_file_name(filename)
```

Comment:
Let's definitely add the comment but I think some spec coverage would be prudent here. These are the types of methods where unit tests make tons of sense. This reminds me of the `Omnibus::Metadata` class:
https://github.com/chef/omnibus/blob/master/lib/omnibus/metadata.rb
https://github.com/chef/omnibus/blob/master/spec/unit/metadata_spec.rb